scheduler/server/stateful_scheduler.go

```diff
@@ -1162,6 +1146,29 @@ func (s *statefulScheduler) processKillJobRequests(reqs []jobKillRequest) {
 	}
 }
 
+// abortTask abort a task - will be triggered by killing a job and when the scheduling algorithm rebalances the workers
+func (s *statefulScheduler) abortTask(jobState *jobState, task *taskState, logFields log.Fields) []saga.SagaMessage {
+	logFields["taskID"] = task.TaskId
+	log.WithFields(logFields).Info("aborting task")
+	var updateMessages []saga.SagaMessage
+
+	if task.Status == domain.InProgress {
+		task.TaskRunner.Abort(true, UserRequestedErrStr)
+	} else if task.Status == domain.NotStarted {
+		st := runner.AbortStatus("", tags.LogTags{JobID: jobState.Job.Id, TaskID: task.TaskId})
+		st.Error = UserRequestedErrStr
```

Comment:
It's not UserRequested

Reply:
thanks! made into parameter.  Rebalance aborts will use "RebalanceRequested"